bpf/lib/nodeport.h

```diff
@@ -992,21 +996,23 @@ static __always_inline int rev_nodeport_lb6(struct __ctx_buff *ctx, __u32 *ifind
 	__u32 monitor = 0;
 	bool l2_hdr_required = true;
 
-	if (!revalidate_data(ctx, &data, &data_end, &ip6))
-		return DROP_INVALID;
-
-	tuple.nexthdr = ip6->nexthdr;
-	ipv6_addr_copy(&tuple.daddr, (union v6addr *) &ip6->daddr);
-	ipv6_addr_copy(&tuple.saddr, (union v6addr *) &ip6->saddr);
+	ret = ipv6_extract_tuple(ctx, &tuple, &l4_off);
+	if (ret == DROP_CT_UNKNOWN_PROTO)
+		goto out;
+	if (IS_ERR(ret))
+		return ret;
 
-	hdrlen = ipv6_hdrlen(ctx, &tuple.nexthdr);
-	if (hdrlen < 0)
-		return hdrlen;
 #ifdef ENABLE_NAT_46X64_GATEWAY
 	if (nat_46x64_fib)
 		goto skip_rev_dnat;
 #endif
-	l4_off = l3_off + hdrlen;
+
+	if (l4_load_ports(ctx, l4_off, &tuple.dport) < 0)
+		return DROP_CT_INVALID_HDR;
+
+	if (!ct_has_nodeport_egress_entry6(get_ct_map6(&tuple), &tuple))
+		goto out;
```

Comment:
IIUC, only this last commit is a bugfix, so only this commit should be backported to stable branches. Unless I'm missing some dependency between the commits (?), the four previous commits can go into another pull request and don't need to be backported.

Reply:
We could arguably defer patches 2/3. But I believe they're worth having in this series to clarify the current behavior, and self-document what sort of traffic is actually affected by the subsequent changes in the RevDNAT code.

ie. if we missed any condition in the "this needs to continue in `CILIUM_CALL_IPV4_NODEPORT_REVNAT`" check, that's an indication that the RevDNAT changes themselves might also be incomplete.